src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -876,8 +874,8 @@ case class MatrixMapRows(child: MatrixIR, newRow: IR) extends MatrixIR {
     val localRowType = prev.typ.rvRowType
     val localEntriesType = prev.typ.entryArrayType
     val entriesIdx = prev.typ.entriesIdx
-    val localGlobalsType = typ.globalType
-    val localColsType = TArray(typ.colType)
+    val localGlobalsType = prev.typ.globalType
```

Comment:
correct, but my change makes it clearer what's going on. I noticed it because I copied and pasted the code for MapColsRow and for obvious reasons, it caused errors.

Reply:
yep, agreed.